### ISSUE TITLE
Rebuild for Angular 5, TypeScript, and npm

### ISSUE THREAD

**Issue by sProject** [Mottie/Keyboard#681]
@Mottie We all love your keyboard in my team. I personally have used it with Angular 1.x and it's performance is unbelievable - by far the most stable thing out there especially on touch screen devices.

However, since we upgraded our stack to ng5 and ts it has become more difficult to work with the keyboard. We are considering rebuilding to make it more user friendly to newer angular stacks and thus promote your work even more.

I'd like to ask for your general opinion on this topic, may be you know someone that is already doing this. Also I'm sure that we'll need some guidance here and there, do you mind if we trouble you from time to time when we get blocked by something.

Thanks

**Comment by Mottie** [comment]
Hi @sProject!

Sure, I wouldn't mind helping out if I can. Sadly, I have dove into learning angular so I may not be too helpful. If you need to chat, I am on slack, discord (`Mottie#0026`) and sometimes [gitter.im](https://gitter.im/Mottie/Keyboard).

**Comment by tudorciotlos** [comment]
@sProject please keep us updated with you're progress regarding the Angular 2+ virtual keyboard. If you have a public repository for the keyboard project, I'm sure more people would be happy to contribute.

**Comment by mattsmith-au** [comment]
I've been working with this great keyboard in an Angular 2+ (6.1.4) application recently.

@sProject What are some of the difficulties your team is experiencing? I don't quite follow why you feel it requires a rebuild. Or are you trying to avoid the jQuery dependency in your Angular app?

**Comment by joaquincrippa** [comment]
Hello @mattsmith-au, I'm having an issue with the two-way binding. I type some text in an input, using the virtual-keyboard, but then I print the variable using console.log and the variable has not been updated. I'm using Angular 6.1.0. Thanks in advance!

**Comment by mattsmith-au** [comment]
@joaquincrippa I had to work through the same issue. Angular's `[(ngModel)]` relies on an 'input' type Event being dispatched from the input element when its value is changed. When interacting with the virtual keyboard text is inserted into the input, but no event is fired. The keyboard will also capture certain interactions with the physical keyboard (meaning hitting backspace will update the text within the input but Angular is unaware of the change, hence the backing variable is not updated).

To work around this issue I hooked into the keyboards [change callback](https://github.com/Mottie/Keyboard/wiki/Methods#change---function) and manually dispatched an 'input' event from the input element whenever a change occurs. 

I'm not sure how you're wiring up the keyboard but our app does the following:

```
$(this._elementRef.nativeElement).keyboard({
  ...
  change: (event, keyboard, el) => {
    this._elementRef.nativeElement.dispatchEvent(new Event('input'));
  },
  ...
});
```
